### DISCUSSION TITLE
token-spray error

### DISCUSSION THREAD

**Discussion by VinzelOps** [projectdiscovery/nuclei#3164] [Q&A] [answered]
[WRN] [api-zenrows] Could not make http request for https://api.zenrows.com/v1/?apikey={{token}}&url=https://example.com: unresolved variables found: token
[WRN] [api-wordnik] Could not make http request for https://api.wordnik.com/v4/word.json/hedgehog/topExample?useCanonical=false&api_key={{token}}: unresolved variables found: token
[WRN] [api-mailgun] Could not make http request for https://api.mailgun.net/v3/domains: unresolved variables found: base64('api:' + token)
[WRN] [api-alienvault] Could not make http request for https://otx.alienvault.com/api/v1/pulses/subscribed/api/v1/pulses/subscribed?page=1&page=1: unresolved variables found: token
[WRN] [api-asana] Could not make http request for https://app.asana.com/api/1.0/users/me: unresolved variables found: token
[INF] No results found. Better luck next time!

Please for help

**Comment by ehsandeep** [comment, marked as answer]
@VinzelOps those are warning and not errors, token-spray requires token input at run time, otherwise, they will get skipped as you can see, more details on running token spray template - https://github.com/projectdiscovery/nuclei-templates/blob/main/token-spray/README.md
